lib/matplotlib/pyplot.py

```diff
@@ -1054,30 +1054,30 @@ def subplots(nrows=1, ncols=1, sharex=False, sharey=False, squeeze=True,
         Number of columns of the subplot grid.
 
     sharex : bool or string, optional, default: False
-        - If True, the X axis will be shared amongst all subplots. Note that
-          if nrows > 1 then the x tick labels won't be displayed on any of
-          plots but the ones on the bottom row.
+        - If True, the X axis will be shared amongst all subplots.
         - If False, no axis will be shared amongst subplots.
         - If a string must be one of "row", "col", "all", or "none".
             - "all" has the same effect as True.
             - "none" has the same effect as False.
             - If "row", each subplot row will share a X axis.
-            - If "col", each subplot column will share a X axis. Note that if
-              nrows > 1 then the x tick labels won't be displayed on any of
-              the plots but the ones on the bottom row.
+            - If "col", each subplot column will share a X axis.
+
+        Note that if the x-axis is shared across rows (sharex=True or
+        sharex="col") and nrows > 1 then the x tick labels won't be displayed
+        on any of plots but the ones on the bottom row.
```

Comment:
Upon further reflection, I think you can state this more simply as "Note that if the x-axis is shared across rows (...), then the x tick labels will only be displayed on the bottom row" (skipping the remark about nrows > 1).
(By the way let's be consistent between "x-axis" and "X axis" (I prefer the former)).